### PR TITLE
Feature/add todo

### PR DESCRIPTION
**Pull request by afm-wtag**

### CODE REVIEW

File: app.js
Comment (hhu-wtag) [inline app.js:1]: 1. Let's start to use a central place to store our todos. 
2. We'll first make any changes to any todo in the todos list. Then we would re-render the todos list.

```javascript
let todos = []

function renderTodos(){
  // iterate over the todos list, create specific todos, and insert them in the DOM
}

function createTodo(todo){
  // responsible for creating a todo element based on the todo object
}

function onAddTodo(){
  // responsible for delegating the work needed to create a todo object
  // for example,
  const todo = {
    title: $todoInput.value,
    createdAt: Date.now()
    //... other options might be added in the future
  }
  
  todos.push(todo)
  
  renderTodos()
}
```

File: app.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const taskManager = () => {
+  const inputTodo = document.getElementById("inputTodo");
+  const addTodoButton = document.getElementById("addTodoButton");
+  const listTodo = document.getElementById("listTodo");
```
Comment (hhu-wtag) [inline app.js]: To separate variables containing DOM elements, we can follow the convention of adding `$` sign as a prefix to the variables that contains DOM elements.

```suggestion
  const $todoInput = document.getElementById("inputTodo");
  const $addTodoButton = document.getElementById("addTodoButton");
  const $todoList = document.getElementById("listTodo");
```

```js
const todo = {}
const $todo = document.querySelector('.todo')
```

As you can see, by just looking at the variable name, I can instantly say which one holds the todo DOM element and which on has the todo state

File: app.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const addTodo = () => {
```
Comment (hhu-wtag) [inline app.js]: As suggested on the file review, this function will be responsible for delegating work
```suggestion
  const onAddTodo = () => {
```

File: app.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    const inputText = inputTodo.value.trim();
+    if (inputText == "") {
+      alert("Enter Something!");
```
Comment (hhu-wtag) [inline app.js]: Don't use I/O blocking calls like `alert`. This is a bad UX practice. Instead, we will figure out a way to show errors to users.
```suggestion
```

File: index.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>TODO App using JS</title>
+  </head>
+  <body>
+    <input id="inputTodo" type="text" placeholder="Enter your todo" />
+    <button id="addTodoButton">Add Todo</button>
+    <ul id="listTodo"></ul>
```
Comment (hhu-wtag) [inline index.html]:
```suggestion
    <div class="todo-input-container">
      <input id="inputTodo" type="text" placeholder="Enter your todo" />
      <button id="addTodoButton">Add Todo</button>
    </div>
        
    <ul id="listTodo"></ul>
```

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: app.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+const taskManager = () => {
+  let todos = [];
+  const $todoInput = document.getElementById("inputTodo");
+  const $addTodoButton = document.getElementById("addTodoButton");
```
Comment (hhu-wtag) [inline app.js]:
```suggestion
  let todos = [];
  
  const $todoInput = document.getElementById("inputTodo");
  const $addTodoButton = document.getElementById("addTodoButton");
```

File: app.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  const $todoList = document.getElementById("listTodo");
+  const $feedbackMessage = document.getElementById("feedbackMessage");
+  function renderTodos() {
+    // iterate over the todos list, create specific todos, and insert them in the DOM
```
Comment (hhu-wtag) [inline app.js]: Have a look at this suggestion: https://github.com/hhe-wtag/dom-todo/pull/1#discussion_r1795185766
```suggestion
```

File: app.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    const $todoText = document.createElement("span");
+    $todoText.textContent = todo.title;
+    $li.appendChild($todoText);
+    return $li;
+  }
+  function showFeeback() {
+    $feedbackMessage.innerText = "Please Enter a valid Todo item";
+    $feedbackMessage.style.color = "red";
+    $feedbackMessage.style.fontSize = "12px";
+  }
```
Comment (hhu-wtag) [inline app.js]: Good thinking.
Let's make this function generic. 
1. Move this to a utility.js file
2. Take the text content as parameter

```suggestion
  function showFeeback(content) {
    $feedbackMessage.innerText = content;
    $feedbackMessage.style.color = "red";
    $feedbackMessage.style.fontSize = "12px";
  }
```

File: app.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  function clearFeedback() {
+    $feedbackMessage.innerText = "";
+  }
+  function onAddTodo() {
+    // responsible for delegating the work needed to create a todo object
+    // for example,
+    const inputText = $todoInput.value.trim();
+    if (inputText == "") {
+      showFeeback();
```
Comment (hhu-wtag) [inline app.js]:
```suggestion
      showFeeback("Please add todo text!");
```

File: app.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      return;
+    }
+    clearFeedback();
+    const todo = {
```
Comment (hhu-wtag) [inline app.js]: Use spacing between statements
```suggestion
    clearFeedback();
    
    const todo = {
```

File: app.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+      title: $todoInput.value,
+      createdAt: Date.now(),
+      //... other options might be added in the future
```
Comment (hhu-wtag) [inline app.js]:
```suggestion
```

File: app.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    };
+    todos.push(todo);
+    renderTodos();
+    $todoInput.value = "";
```
Comment (hhu-wtag) [inline app.js]: ditto. 
ditto means, apply some reviews that were given earlier. In this case, use spacing.
```suggestion
    todos.push(todo);
    
    renderTodos();
    
    $todoInput.value = "";
```

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: app.js
Comment (afm-wtag) [reply app.js:1]: Thank u so much vaiya for the suggestion , updated the code according to it.

File: app.js (hunk 2 above)
Comment (afm-wtag) [reply app.js]: function naming updated

File: app.js (hunk 3 above)
Comment (afm-wtag) [reply app.js]: alert is removed

File: index.html (hunk 4 above)
Comment (afm-wtag) [reply index.html]: html code updated

File: app.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+  function showFeeback(content) {
+    $feedbackMessage.innerText = content;
+    $feedbackMessage.style.color = "red";
+    $feedbackMessage.style.fontSize = "12px";
+  }
+
+  function clearFeedback() {
+    $feedbackMessage.innerText = "";
+  }
```
Comment (hhu-wtag) [inline app.js]: Move these functions to `utility.js` file and import it from there

**Review by hhu-wtag — APPROVED** [review]

File: app.js (hunk 12 above)
Comment (afm-wtag) [reply app.js]: Updated the code in utility.js file

**Pull request merged**